shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -754,28 +754,52 @@ - (void)onUserSettingsChanged:(NSNotification*)notification {
 
 - (CGFloat)textScaleFactor {
   UIContentSizeCategory category = [UIApplication sharedApplication].preferredContentSizeCategory;
-  // The delta is computed based on the following:
-  // - L (large) is the default 1.0 scale.
-  // - The scale is linear spanning from XS to XXXL.
-  // - XXXL = 1.4 * XS.
+  // The delta is computed by approximating Apple's typography guidelines:
+  // https://developer.apple.com/ios/human-interface-guidelines/visual-design/typography/
   //
-  // L    = 1.0      = XS + 3 * delta
-  // XXXL = 1.4 * XS = XS + 6 * delta
-  const CGFloat delta = 0.055555;
+  // Specifically:
+  // Non-accessibility sizes for "body" text are:
+  const CGFloat xs = 14;
+  const CGFloat s = 15;
+  const CGFloat m = 16;
+  const CGFloat l = 17;
+  const CGFloat xl = 19;
+  const CGFloat xxl = 21;
+  const CGFloat xxxl = 23;
+
+  // Accessibility sizes for "body" text are:
+  const CGFloat ax1 = 28;
+  const CGFloat ax2 = 33;
+  const CGFloat ax3 = 40;
+  const CGFloat ax4 = 47;
+  const CGFloat ax5 = 53;
+
+  // We compute the scale as relative difference from size L (large, the default size), where
+  // L is assumed to have scale 1.0.
   if ([category isEqualToString:UIContentSizeCategoryExtraSmall])
-    return 1.0 - 3 * delta;
+    return xs / l;
```

Comment:
Are these first three missing a minus sign? They used to be negative...

Reply:
They were positive, but <1.0, which is still the case. The only change here is that instead of scaling linearly by adding so many "deltas", I compute the scale based on ratios of font sizes defined by Human Interface Guidelines.

Reply:
I read the delta as 0.5 instead of 0.05 and that's how I thought it's negative. Never mind then.